plugins/woocommerce-blocks/assets/js/blocks/add-to-cart-with-options/variation-selector/edit.tsx

```diff
@@ -0,0 +1,76 @@
+/**
+ * External dependencies
+ */
+import { useBlockProps } from '@wordpress/block-editor';
+import { BlockEditProps } from '@wordpress/blocks';
+import { useProductDataContext } from '@woocommerce/shared-context';
+import {
+	// @ts-expect-error Using experimental features
+	// eslint-disable-next-line @wordpress/no-unsafe-wp-apis
+	__experimentalHeading as Heading,
+	SelectControl,
+} from '@wordpress/components';
+
+interface Attributes {
+	className?: string;
+}
+
+const AddToCartWithOptionsVariationSelectorEdit = (
+	props: BlockEditProps< Attributes >
+) => {
+	const { className } = props.attributes;
+	const { product } = useProductDataContext();
+
+	const blockProps = useBlockProps( {
+		className,
+	} );
+
+	if ( product.type !== 'variable' ) {
+		return null;
+	}
+
+	const renderAttributeSelectors = () => {
+		return Object.entries( product.attributes ).map(
+			( [ attributeKey, attribute ] ) => (
+				<div
+					className="wc-block-components-variation-selector__wrapper"
+					key={ attributeKey }
+				>
+					<Heading
+						className="wc-block-components-variation-selector__label"
+						level="3"
+					>
+						{ attribute.name }
+					</Heading>
+					<SelectControl
+						id={ `pa_${ attribute.taxonomy }` }
+						value=""
+						options={ [
+							{
+								label: 'Choose an option',
```

Comment:
We should probably make this string localizable.

Reply:
Fixed in the commit [9905d85](https://github.com/woocommerce/woocommerce/pull/53945/commits/9905d857e3128f946cce50e475c0c761a301c222).